build.sc

```diff
@@ -844,9 +844,23 @@ trait CliIntegration extends SbtModule with ScalaCliPublishModule with HasTests
       def test(args: String*) = {
         val argsTask = T.task {
           val launcher = launcherTask().path
+          val debugReg = "--debug(:[0-9]+)?".r
+          val debugOpt = args.filter(a => debugReg.pattern.matcher(a).matches).headOption
```

Comment:
`filter(...).headOption` is the same as `find(...)`
also, you can figure out the port in a single chain.
```suggestion
          val maybeDebugPort = args.find(regex.matches).flatMap { case regex(port) => Option(port).orElse(Some("5005")); case _ => None }
```
BTW there's no need for the `isDebug` boolean value, after the line above you need only check `maybeDebugPort.isDefined` (since we already fallback to `5005`)